packages/react/src/components/UIShell/HeaderMenu.tsx

```diff
@@ -201,25 +264,23 @@ class HeaderMenu extends React.Component {
       isCurrentPage,
       'aria-label': ariaLabel,
       'aria-labelledby': ariaLabelledBy,
-      className: customClassName,
+      className: customClassName = '',
       children,
       renderMenuContent: MenuContent,
       menuLinkName,
-      focusRef, // eslint-disable-line no-unused-vars
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      focusRef,
       onBlur,
       onClick,
       onKeyDown,
       ...rest
     } = this.props;
 
-    const hasActiveDescendant = (childrenArg) =>
-      React.Children.toArray(childrenArg).some(
-        (child) =>
-          child.props.isActive ||
-          child.props.isCurrentPage ||
-          (child.props.children instanceof Array &&
-            hasActiveDescendant(child.props.children))
-      );
+    const hasActiveChildren = React.Children.toArray(children).some((child) => {
+      if (React.isValidElement(child)) {
+        child.props.isActive || child.props.isCurrentPage;
+      }
+    });
```

Comment:
This is a functional change, right? This no longer recursively looks through descendant children. I think we'll want to retain that.